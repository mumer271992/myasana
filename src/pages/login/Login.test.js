import React from 'react';
import { shallow } from 'enzyme';
import Login from './Login';


describe('Login page component test suit', () => {
  let wrapper;
  const signIn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Login signinUser={signIn} />);
  });

  it('Signin page rendered successfully', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Clicking on signin button should call signin function passed in props from Login High Order function', () => {
    const spy = jest.spyOn(wrapper.instance(), 'signin').mockImplementation(jest.fn());
    wrapper.instance().forceUpdate();
    wrapper.find('[data-test="signin-form"]').simulate('submit');
    expect(spy).toHaveBeenCalled();
  });
});
