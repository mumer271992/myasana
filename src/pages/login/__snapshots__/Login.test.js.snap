// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page component test suit Signin page rendered successfully 1`] = `
<div
  className="page login-page"
>
  <h2
    className="text-center logo"
  >
    My Asana
  </h2>
  <div
    className="form-container"
  >
    <h2
      className="text-center"
    >
      Login
    </h2>
    <form
      data-test="signin-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label>
          Email address
        </label>
        <input
          aria-describedby="emailHelp"
          className="form-control"
          id="exampleInputEmail1"
          name="email"
          onChange={[Function]}
          placeholder="Enter email"
          type="email"
        />
        <small
          className="form-text text-muted"
          id="emailHelp"
        >
          We'll never share your email with anyone else.
        </small>
      </div>
      <div
        className="form-group"
      >
        <label>
          Password
        </label>
        <input
          className="form-control"
          id="exampleInputPassword1"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Login
      </button>
    </form>
    <Link
      replace={false}
      to="/signup"
    >
      <button
        className="btn btn-default"
        type="submit"
      >
        Sign Up
      </button>
    </Link>
  </div>
</div>
`;
